EsApi4DScheduleSampleApp/EsApi4DScheduleSampleApp/Program.cs

```diff
@@ -0,0 +1,119 @@
+        }
+        else
+        {
+            ConsoleApp.Log("No items for single schedule.");
+        }
+        return;
+    }
+    else if (arguments.Post)
+    {
+        ConsoleApp.Log("Fetching all resource status history - Use this endpoint to query all resource status history items in the specified project schedule.");
+        ConsoleApp.Log($"Sending GET request to {client.BaseAddress}{arguments.Schedule}/resourceStatusHistory");
+        var get = new HttpGet($"{arguments.Schedule}/resourceStatusHistory", client);
+        var response = get.Get().Result;
+        var stringManipulated = response.Remove(0, 16);
+        var stringSplit = stringManipulated.Split('\"');
+        Console.WriteLine();
+
+        if (!response.Contains("\"items\":[]"))
+        {
+            ConsoleApp.Log("Posting new resource status history - Add a new resource status history item to an associated resource.");
+            ConsoleApp.Log($"Sending POST request to {client.BaseAddress}{arguments.Schedule}/resourceStatusHistory");
+            var post = new HttpPost($"{arguments.Schedule}/resourceStatusHistory", client, stringSplit);
+            var postResp = post.Post().Result;
+        }
+        else
+        {
+            ConsoleApp.Log("No items in Resource Status History. Skipping individual ID endpoint requests.");
+        }
+        return;
+    }
+    else
+    {
+        ConsoleApp.Log("Fetching all resource status histories - Use this endpoint to query all resource status history items in the specified project schedule.");
+        ConsoleApp.Log($"Sending GET request to {client.BaseAddress}{arguments.Schedule}/resourceStatusHistory");
+        var get = new HttpGet($"{arguments.Schedule}/resourceStatusHistory", client);
+        var response = get.Get().Result;
+        var stringManipulated = response.Remove(0, 16);
+        var stringSplit = stringManipulated.Split('\"');
+        var id = stringSplit[1];
+
+        if (!response.Contains("\"items\":[]"))
+        {
+            ConsoleApp.Log("Fetching single resource status history - Use this endpoint to query a single resource status history item in the specified project schedule.");
+            ConsoleApp.Log($"Sending GET request to {client.BaseAddress}{arguments.Schedule}/resourceStatusHistory/{id}");
+            get.RequestUri = $"{arguments.Schedule}/resourceStatusHistory/{id}";
+            response = get.Get().Result;
+
+            ConsoleApp.Log("Posting new resource status history - Add a new resource status history item to an associated resource.");
+            ConsoleApp.Log($"Sending POST request to {client.BaseAddress}{arguments.Schedule}/resourceStatusHistory");
+            var post = new HttpPost($"{arguments.Schedule}/resourceStatusHistory", client, stringSplit);
+            var postResp = post.Post().Result;
+
+            if (postResp.Contains("changeRequestId"))
+            {
+                stringSplit = postResp.Split('\"');
```

Comment:
Why are you manually extracting data by manipulating strings instead of using JSON deserialization? Do you expect your API users to do the same? If so it's very brittle because any additions to your responses could break things.

Reply:
changed to deserialisation in most recent commit